js/main.js

```diff
@@ -61,13 +61,13 @@ function setWeatherLocation(location) {
 }
 
 function alternateIcon(event) {
-  if (event.target && event.target.nodeName === 'LI') {
-    if (event.target.children[0].className === 'far fa-circle') {
-      event.target.children[0].className = 'far fa-check-circle';
-      data.template[data.weatherOptions[event.target.value]] = true;
-    } else if (event.target.children[0].className === 'far fa-check-circle') {
-      event.target.children[0].className = 'far fa-circle';
-      data.template[data.weatherOptions[event.target.value]] = false;
+  if (event.target && (event.target.nodeName === 'BUTTON' || event.target.nodeName === 'I')) {
+    if (event.target.closest('.weather-information-list-choice').children[0].children[0].className === 'far fa-circle') {
+      event.target.closest('.weather-information-list-choice').children[0].children[0].className = 'far fa-check-circle';
+      data.template[data.weatherOptions[event.target.closest('.weather-information-list-choice').value]] = true;
+    } else if (event.target.closest('.weather-information-list-choice').children[0].children[0].className === 'far fa-check-circle') {
+      event.target.closest('.weather-information-list-choice').children[0].children[0].className = 'far fa-circle';
+      data.template[data.weatherOptions[event.target.closest('.weather-information-list-choice').value]] = false;
```

Comment:
It's not good to do stuff like `.children[0].children[0]`. this will break immediately or in a weird way if you need to restructure your document. Instead, query the element for a descendent.

```js
const $listChoice = event.target.closest('.weather-information-list-choice')
const $circle = $listChoice.querySelector('fa-circle');

if ($circle.classList.contains('fas')) {

} else {

}
```

See if you can clean this up in your other code too.

https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelector
https://developer.mozilla.org/en-US/docs/Web/API/Element/classList